### ISSUE TITLE
Carry over fig.alt label when referencing chunks with ref.label

### ISSUE THREAD

**Issue by mine-cetinkaya-rundel** [yihui/knitr#1959]
This is a feature request to carry over the `fig.alt` option when referencing chunks with `ref.label`:

When earlier chunks are referenced with `ref.label`, chunk options don't carry along to the next chunk. It would be nice if, at a minimum, the `fig.alt` text was carried over to avoid the need for copying and pasting the alt text (which can sometimes be extensive). In the following example, the same figure is shown twice, once without and once with the code. In the rendered document alt text only appears for the first plot, not the second.

```
---
output: html_document
---

```{r pressure, echo = FALSE, fig.alt = "This is alt text for pressure plot."}
plot(pressure)
` ``


```{r ref.label = "pressure"}
` ``
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('knitr')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('yihui/knitr')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by cderv** [comment]
For reference, summarizing here the discussion we had: 

* `ref.label` only use the code part and the not the option as usually you wan't to change them. Options are only local
* It exists currently a mechanism to reuse chunk options in multiple place with [opts_template](https://bookdown.org/yihui/rmarkdown-cookbook/opts-template.html) but it won't cover this use case I believe.
* [Text reference feature](ttps://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#text-references) in bookdown is one use case for this. But `fig.alt` should still be set in the chunk that show the code. 

I feel this need is specific to Figures, and we may need a special "thing" for figures. 

* Like a figure engine that would behave differently with fig.* option moved ?
* Or another way to define fig options for figure chunks. This new way could take the labels of which to apply to - and the option would apply on all the chunks of the same label. 🤔 

This could be a need for other local option `fig.*` and not just `fig.alt`

Thanks for opening the FR here @mine-cetinkaya-rundel !

**Comment by atusy** [comment]
@mine-cetinkaya-rundel This is also what I wanted!! I implemented it in PR #1960.
Current implementation introduces a helper function `with_opts`.
You can try it by installing `remotes::install_github("yihui/knitr#1960")`

````
---
output: html_document
---

```{r pressure, echo = FALSE, fig.alt = "This is alt text for pressure plot."}
plot(pressure)
```


```{r ref.label = knitr::with_opts("pressure")}
```

````

**Comment by mine-cetinkaya-rundel** [comment]
That is very neat, thank you @atusy!
